core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/sync/operationresult/CmSyncOperationResultFactory.java

```diff
@@ -0,0 +1,11 @@
+package com.sequenceiq.cloudbreak.service.upgrade.sync.operationresult;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class CmSyncOperationResultFactory {
```

Comment:
I don't see any benefit for creating this factory. Could you delete it or explain why is it necesary?

Reply:
Calling new is a static method, and makes testing harder. In tests I do not like to add wiring for any(), but for a concrete value. The other option to achieve this would have been introducing equals and hashCode for the class created here and so for all contained classes. But equals then also needs some testing (although is autogenerated, so should work), I thought it is easier just to move creation out.
I also checked before extracting that there are some examples of such very simple factories in the cloudbreak repo.

Reply:
I haven't check the test, but you have to test the interfaces of the class. I don't think making the code more complicated to make testing easier is a good idea.